### PR TITLE
[FIX]: Enforce LF line endings via .gitattributes

### PR DESCRIPTION
**Pull request by MohamadNazik**
## Summary

This PR fixes a critical issue where shell scripts (`.sh`) are automatically converted from LF to CRLF line endings on Windows machines. This conversion breaks the scripts when they are executed inside Linux-based Docker containers (e.g., `thunder-setup`). We have introduced a `.gitattributes` file to enforce **LF** line endings repository-wide for all shell scripts, ensuring they remain executable across all platforms.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## Changes Made

- **Added** `.gitattributes` file in the root directory to enforce `eol=lf` for `*.sh` files.
- **Normalized** all existing shell scripts to use LF line endings and UTF-8 (No BOM) encoding.
- **Fixed** potential syntax errors in `idp/` scripts caused by incorrect line-ending interpretations in the Linux environment.

## Testing

- [x] I have tested this change locally (verified `.\start-dev.ps1 -CleanRun` completes successfully on Windows).
- [x] I have tested edge cases (verified that Git stash/pop and branch switching no longer reverts scripts to CRLF).
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #444 

## Screenshots/Demo

N/A

## Additional Notes

The use of UTF-8 (without BOM) is essential to preserve the emojis and special symbols used in the bootstrap scripts while maintaining compatibility with the Linux shell.

## Deployment Notes

No special deployment steps required; Git will automatically apply the new line-ending policy to existing clones upon the next pull/checkout.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,7 @@
+* text=auto
```
Comment (sthanikan2000) [inline .gitattributes]:
```suggestion
* text=auto eol=lf
```

Let's enforce LF everywhere by default

**Review by sthanikan2000 — CHANGES_REQUESTED** [review]

**Review by sthanikan2000 — DISMISSED** [review]

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,7 @@
+* text=auto eol=lf
+
+# Enforce LF line endings for all shell scripts to ensure compatibility with Linux containers
+*.sh text eol=lf
+
+# Enforce LF for env files as they are sourced by shell scripts
+.env* text eol=lf
```
Comment (mushrafmim) [inline .gitattributes]:
```suggestion
```
After adding for all `*`, why do we need to add specifically again, aren't these files already covered.

**Review by mushrafmim — CHANGES_REQUESTED** [review]

**Review by mushrafmim — DISMISSED** [review]

**Review by mushrafmim — APPROVED** [review]

**Review by sthanikan2000 — APPROVED** [review]
